sphinx_gallery/docs_resolv.py

```diff
@@ -42,15 +42,15 @@ def _get_data(url):
             encoding = resp.headers.get('content-encoding', 'plain')
         data = resp.read()
         if encoding == 'plain':
-            pass
+            data = data.decode('utf-8')
         elif encoding == 'gzip':
             data = StringIO(data)
-            data = gzip.GzipFile(fileobj=data).read()
+            data = gzip.GzipFile(fileobj=data).read().decode('utf-8')
         else:
             raise RuntimeError('unknown encoding')
     else:
-        with open(url, 'r') as fid:
-            data = fid.read()
+        with open(url, mode='rb') as fid:
```

Comment:
It would be better to use `codecs.open(..., encoding='utf-8')` to be consistent with what we are doing elsewhere in the code.